Add a user documentation for concrete-core (#125)
zama-ai/concrete: concrete-core/docs/user/introduction.md

# Introduction

> Welcome to the `concrete-core` guide !

Reviewer: In english, I think there is no space before "!"

Reviewer: This comment applies many times in the files, but I post it just once to avoid being too repetitive.

This library contains a set of fast, low-level primitives which can be used to implement *Fully

> Homomorphically Encrypted* (FHE) programs. In a nutshell, fully homomorphic encryption makes it possible to

Reviewer: Fully Homomorphically Encrypted -> Fully Homomorphically Encryption

Reviewer: "In a nutshell, fully homomorphic encryption makes it" -> "In a nutshell, FHE makes it"

engine.destroy(plaintext).unwrap();
engine.destroy(ciphertext).unwrap();
engine.destroy(decrypted_plaintext).unwrap();
```

## Audience

The goal of `concrete-core` is to be the intermediate layer where library writers and compiler makers, can
access a breadth of fast primitives developed by researchers and hardware designers. As such, this guide
contains two kinds of tutorials geared towards this two categories of users.

### Contributors to the public API

If you are a researcher working on a fast FHE algorithm, or a hardware designer who want to accelerate an
existing algorithm, you can contribute your code to `concrete-core`. What this gives you is:
+ The `concrete-benchmark` application, which allows to benchmark your implementation in minutes, and compare
the results with the existing implementations.
+ The `concrete-test` application, which allows to easily test your implementation, to verify that your
implementation is correct.
+ Exposure to external users, which can easily access your implementation, and integrate it within their
library or compiler.

If you are interested in contributing new algorithms, or new hardware accelerations to `concrete-core`,

> proceed to the [Contributor Guide](architecture.md) section !

Reviewer 2: ```suggestion
proceed to the [Contributor Guide](../dev/index.rst) section !
```

Reviewer 3: Done !

### Consumers of the public API

> If you are a writing a library or a compiler for FHE, you can rely on the public API of `concrete-core` to

Reviewer 2: ```suggestion
If you are writing a library or a compiler for FHE, you can rely on the public API of `concrete-core` to
```

access a large set of homomorphic operators. To properly use those operators though, you have to know your way
around FHE. The API gives you the freedom to control multiple parameters, which can lead to less than 128
bits of security if chosen incorrectly. As a rule of thumb, if you did not carefully study the mathematical
foundations of FHE, `concrete-core` is probably too low-level for you to use. Fortunately, we propose multiple
libraries that build on top of `concrete-core` and which propose a safer API. To see which one best suits your

> needs, see the [concrete homepage](https://zama.ai/concrete).

Reviewer: "To see which one best suits your needs" -> "To see which one suits your needs best"? Not sure

If you are interested of building a higher level abstraction using the `concrete-core` api, proceed to the

> [User Guide](backends.md) section !

Reviewer 2: ```suggestion
[API](../api/backends.md) section!
```

Reviewer 3: Fixed thanks !
